core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1139,9 +1139,13 @@ object Log {
       throw exception(dir)
 
     val name: String =
-      if (dirName.endsWith(DeleteDirSuffix)) dirName.substring(0, dirName.indexOf('.'))
+      if (dirName.endsWith(DeleteDirSuffix)) dirName.substring(0, dirName.lastIndexOf('.'))
       else dirName
 
+    if (!name.contains('-')) {
```

Comment:
Thanks. Maybe we should cherry-pick the commit that added that check in trunk to 0.10.2 as well.